scripts/ec2/ls.coffee

```diff
@@ -29,29 +30,36 @@ DAYS_CONSIDERED_SOON = 2
 ec2 = require('../../ec2.coffee')
 
 getArgParams = (arg, filter = "all", opt_arg = "") ->
-  instances = []
+  filterValues = []
   if arg
     for av in arg.split /\s+/
       if av and not av.match(/^--/)
-        instances.push(av)
+        filterValues.push(av)
+
+  #console.log "filterValues are: "
+  #console.log util.inspect(filterValues, false, null)
 
   params = {}
 
   if filter == "mine"
     params['Filters'] = [{Name: 'tag:Creator', Values: [opt_arg]}]
   else if filter == "chat"
     params['Filters'] = [{Name: 'tag:CreatedByApplication', Values: [filter]}]
-  else if filter == "filter" and instances.length
-    params['Filters'] = [{Name: 'tag:Name', Values: ["*#{instances[0]}*"]}]
-  else if instances.length
-    params['InstanceIds'] = instances
+  else if filter == "filter" and filterValues.length
+    params['Filters'] = [{Name: 'tag-value', Values: ["*#{filterValues[0]}*"]}]
```

Comment:
using `tag-value` will cause it to apply the search filter across any tags for instances. Probably the single most important change in this PR